core/services/bulletprooftxmanager/errors.go

```diff
@@ -0,0 +1,128 @@
+package bulletprooftxmanager
+
+import (
+	"regexp"
+	"strings"
+
+	"github.com/pkg/errors"
+)
+
+// fatal means this transaction can never be accepted even with a different nonce or higher gas price
+type sendError struct {
+	fatal bool
+	err   error
+}
+
+func (s *sendError) Error() string {
+	return s.err.Error()
+}
+
+func (s *sendError) StrPtr() *string {
+	e := s.err.Error()
+	return &e
+}
+
+// Fatal indicates whether the error should be considered fatal or not
+// Fatal errors mean that no matter how many times the send is retried, no node
+// will ever accept it
+func (s *sendError) Fatal() bool {
+	return s != nil && s.fatal
+}
+
+// Parity errors
+var (
+	// Non-fatal
+	parTooCheapToReplace    = regexp.MustCompile("^Transaction gas price .+is too low. There is another transaction with same nonce in the queue")
```

Comment:
typo here!

Reply:
Where?

Reply:
I misread that line, it's a regexp, not a typo.